test/test_diffusion.py

```diff
@@ -319,43 +424,72 @@ def test_diffusion_accuracy(actx_factory, problem, nsteps, dt, scales, order,
 
         nodes = actx.thaw(dcoll.nodes())
 
+        if error_boundary:
+            target_boundary = error_boundary[0]
+            value = error_boundary[1]
+            alpha = error_boundary[2]
+        else:
+            value = None
+            alpha = None
+
         def get_rhs(t, u):
             kappa = p.get_kappa(nodes, t, u)
             if isinstance(kappa, DOFArray):
                 quadrature_tag = DISCR_TAG_QUAD
             else:
                 quadrature_tag = DISCR_TAG_BASE
+            boundaries = p.get_boundaries(dcoll, actx, t, value=value, alpha=alpha)
             return (
-                diffusion_operator(
-                    dcoll, kappa=kappa, boundaries=p.get_boundaries(dcoll, actx, t),
-                    u=u, quadrature_tag=quadrature_tag)
-                + evaluate(sym_f, x=nodes, t=t))
+                diffusion_operator(dcoll, kappa=kappa, boundaries=boundaries,
+                                   u=u, quadrature_tag=quadrature_tag)
+                + evaluate(sym_f, x=nodes, t=t)
+                + evaluate(sym_source, x=nodes, t=t, u=u)
```

Comment:
FYI, this test is set up to support manufactured solutions in addition to exact solutions; if the solution specified by the problem is not an exact solution, the RHS source term will be computed symbolically (that's the `sym_f` above). So the `sym_source`/`get_source_term` stuff shouldn't be needed.

Reply:
I added the `source`  as an actual source term from the physics of the problem. But with your test suggestion, that is not needed. Removed/aborted adding it everywhere.